### PR TITLE
[cli][tamagui - expo-router - stack] Fix Title and Subtitle default styles 

### LINKED ISSUE #158: Fonts are not correctly apply using Tamagui and expo-router on web
Hey !

After a fresh install, it seems that the font family and color are not working on web using:
- expo-router
- tamagui

Working well on iOS and Android

image

## Expected Behavior

The inter Font should be apply to the text and title

## Current Behavior

The browser serve the default font

## Steps to Reproduce

fresh install using this command 
`npx create-expo-stack test-stack --expo-router --tamagui --firebase --import-alias --yarn`

### PR DESCRIPTION
**Pull request by gwenoleR**
## Description

On fresh start using Expo Router + Tamagui, the text styles were not applied on Web.
Moreover, the fonts were loads only on Detail Screen on Native.

This PR should fix this two issue

Before
before

After
after

## Related Issue

Closes #158 

## How Has This Been Tested?

Start a fresh install with 
```
create-expo-stack debug-tamagu-font --expo-router --tamagui --import-alias --yarn
```

Start the project on each platform

### CODE REVIEW

File: cli/src/templates/packages/expo-router/stack/app/_layout.tsx.ejs [hunk 1]
```diff
@@ -1,16 +1,34 @@
 <% if (props.stylingPackage?.name === "nativewind") { %>
 	import { Stack } from "expo-router";
 <% } else if (props.stylingPackage?.name === "tamagui") { %>
-	import { TamaguiProvider } from "tamagui";
-	import { Stack } from "expo-router";
+  import { Stack } from "expo-router";
```
Comment (danstepanov) [inline cli/src/templates/packages/expo-router/stack/app/_layout.tsx.ejs:4]: it looks like `import { Stack } from "expo-router";` is imported regardless of the choice of styling, we should move it out of these if statements.

File: cli/src/templates/packages/expo-router/stack/app/_layout.tsx.ejs [hunk 2]
```diff
@@ -1,16 +1,34 @@
...
 
-	import config from '../tamagui.config'
+  import config from '../tamagui.config';
 <% } else { %>
 	import { Stack } from "expo-router";
 <% } %>
 
 export default function Layout() {
 
+  <% if(props.stylingPackage?.name === "tamagui") {%>
```
Comment (danstepanov) [inline cli/src/templates/packages/expo-router/stack/app/_layout.tsx.ejs:17]:
```suggestion
  <% if (props.stylingPackage?.name === "tamagui") {%>
```

File: cli/src/templates/packages/expo-router/stack/app/_layout.tsx.ejs [hunk 3]
```diff
@@ -1,16 +1,34 @@
...
+  		const [loaded] = useFonts({
+			Inter: require("@tamagui/font-inter/otf/Inter-Medium.otf"),
+			InterBold: require("@tamagui/font-inter/otf/Inter-Bold.otf")
+		});
```
Comment (danstepanov) [inline cli/src/templates/packages/expo-router/stack/app/_layout.tsx.ejs:21]:
```suggestion
  	const [loaded] = useFonts({
			Inter: require("@tamagui/font-inter/otf/Inter-Medium.otf"),
			InterBold: require("@tamagui/font-inter/otf/Inter-Bold.otf")
		});
```

File: cli/src/templates/packages/tamagui/tamagui.config.ts.ejs [hunk 4]
```diff
@@ -45,15 +45,14 @@ const bodyFont = createInterFont();
...
+    color: '#000',
+  });
 
-	export const Subtitle = styled(Text, {
-		color: '#38434D',
-		fontSize: 36,
-	});
+  export const Subtitle = styled(SizableText, {
+    color: '#38434D',
+    size: '$7',
```
Comment (danstepanov) [inline cli/src/templates/packages/tamagui/tamagui.config.ts.ejs:54]: can you elaborate on how this $ notation works? I've never been able to figure it out.

File: cli/src/templates/packages/tamagui/tamagui.config.ts.ejs [hunk 5]
```diff
@@ -73,7 +72,7 @@ const bodyFont = createInterFont();
 		},
 	});
 
-	export const ButtonText = styled(Text, {
+	export const ButtonText = styled(SizableText, {
```
Comment (danstepanov) [inline cli/src/templates/packages/tamagui/tamagui.config.ts.ejs:75]: Also what is the difference between `Text` and `SizableText`?

**Review by danstepanov — COMMENTED** [review]
just a few nits

File: cli/src/templates/packages/tamagui/tamagui.config.ts.ejs (hunk 4 above)
Comment (gwenoleR) [reply cli/src/templates/packages/tamagui/tamagui.config.ts.ejs:54]: The `size` property is a custom property from Tamagui to manage all font properties according to the theme you defined.

It's use for this react properties:
* fontSize
* fontWeight
* lineHeight
* letterSpacing

In this case the token `$7` is define to apply this values: 
* fontSize: 20
* fontWeight: 600
* lineHeight: 30
* letterSpacing: 0

This values can be edited for each font you create for example:

```
{
    "family": "Inter, Helvetica, Arial, sans-serif",
    "lineHeight": {
        "1": 16,
        "2": 21,
        "3": 22,
        "4": 23,
        "5": 26,
        "6": 28,
        "7": 30,
        "8": 33,
        "9": 41,
        "10": 59,
        "11": 69,
        "12": 76,
        "13": 87,
        "14": 109,
        "15": 133,
        "16": 155,
        "true": 23
    },
    "weight": {
        "1": "400",
        "2": "400",
        "3": "400",
        "4": "400",
        "5": "400",
        "6": "400",
        "7": "600",
        "8": "600",
        "9": "600",
        "10": "600",
        "11": "600",
        "12": "600",
        "13": "600",
        "14": "600",
        "15": "600",
        "16": "600",
        "true": "600"
    },
    "letterSpacing": {
        "1": 0,
        "2": 0,
        "3": 0,
        "4": 0,
        "5": 0,
        "6": 0,
        "7": 0,
        "8": 0,
        "9": 0,
        "10": 0,
        "11": 0,
        "12": 0,
        "13": 0,
        "14": 0,
        "15": 0,
        "16": 0,
        "true": 0
    },
    "size": {
        "1": 11,
        "2": 12,
        "3": 13,
        "4": 14,
        "5": 16,
        "6": 18,
        "7": 20,
        "8": 23,
        "9": 30,
        "10": 46,
        "11": 55,
        "12": 62,
        "13": 72,
        "14": 92,
        "15": 114,
        "16": 134,
        "true": 14
    },
    "face": {
        "100": {
            "normal": "Inter, Helvetica, Arial, sans-serif"
        },
        "200": {
            "normal": "Inter, Helvetica, Arial, sans-serif"
        },
        "300": {
            "normal": "Inter Light",
            "italic": "Inter LightItalic"
        },
        "400": {
            "normal": "Inter",
            "italic": "Inter Italic"
        },
        "500": {
            "normal": "Inter",
            "italic": "Inter Italic"
        },
        "600": {
            "normal": "Inter",
            "italic": "Inter MediumItalic"
        },
        "700": {
            "normal": "Inter SemiBold",
            "italic": "Inter SemiBoldItalic"
        },
        "800": {
            "normal": "Inter Bold",
            "italic": "Inter BoldItalic"
        },
        "900": {
            "normal": "Inter Bold",
            "italic": "Inter BoldItalic"
        },
        "normal": {
            "normal": "Inter Regular",
            "italic": "Inter Italic"
        },
        "bold": {
            "normal": "Inter Bold",
            "italic": "Inter BoldItalic"
        }
    }
}
```

Comment (danstepanov) [reply cli/src/templates/packages/tamagui/tamagui.config.ts.ejs:54]: thanks for the explanation! this makes sense :)

File: cli/src/templates/packages/tamagui/tamagui.config.ts.ejs (hunk 5 above)
Comment (danstepanov) [reply cli/src/templates/packages/tamagui/tamagui.config.ts.ejs:75]: Thanks again for the explanation 🙂

**Comment by danstepanov** [comment]
@gwenoleR Is there a reason you made it such that the styles no longer match what the app previously looked like on iOS? I see that the styles are more universally applied but am curious of your intentions of this visual change. We use iOS as the base case for create expo stack (what android and web should look like).

**Comment by danstepanov** [comment]
Credited here: https://github.com/danstepanov/create-expo-stack/pull/156

**Comment by danstepanov** [comment]
I have amended the commit to be attributed to you https://github.com/danstepanov/create-expo-stack/commit/49550e7026a1b87ddb00bf94aada6047dcd509d7

**Pull request closed without merging**